sdk/core/azure-core/src/http/retry_policy.cpp

```diff
@@ -157,7 +157,7 @@ Context inline CreateRetryContext(Context const& parent)
   int retryCount = 0;
   if (parent.HasKey(RetryKey))
   {
-    retryCount = parent[RetryKey].Get<int>() + 1;
+    retryCount = parent.Get<int>(RetryKey) + 1;
```

Comment:
Why do we store number of retries and not the whole RetryOptions under this key?

Reply:
I'm avoiding altering the logic in this change.  If there is a better design for the above problem we can discuss, but it won't be made as part of this PR